distributed/protocol/pickle.py

```diff
@@ -19,6 +23,59 @@
 pickle_types = tuple(pickle_types)
 
 
+@gen.coroutine
+def _tornado_coroutine_sample():
+    yield
+
+def is_tornado_coroutine(func):
+    """
+    Return whether *func* is a Tornado coroutine function.
+    Running coroutines are not supported.
+    """
+    return func.__code__ is _tornado_coroutine_sample.__code__
```

Comment:
It would need `gen.coroutine` to return something else than a wrapper function. Apparently this has been the case since the beginning, see https://github.com/tornadoweb/tornado/commit/58b0dabd285458043bec38a1861372bd93f6ee27